core/rfb.js

```diff
@@ -1543,6 +1872,90 @@
             sock.flush();
         },
 
+        atenKeyEvent: function (sock, keysym, down) {
+            var ks = XK2HID[keysym];
```

Comment:
Right, the browsers have their own systems. Look at xtscancode.js for mapping between that and XT codes. We should probably merge that table with yours.